### PR TITLE
[DNM] Gandarez merge

### PR DESCRIPTION
**Pull request by vujita**
I didn't verify functionality just mainly merging and fixing lint errors

@gandarez Can you pull this into your branch and create a PR with once you verify everything?
Also check all your functionality

### CODE REVIEW

File: src/editors/editor.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 declare interface Editor {
   name: string;
+  displayName: string;
```
Comment (alanhamlett) [inline src/editors/editor.ts]: Let's just have `name` be formatted like `displayName`, then remove `displayName`.

File: src/editors/xcode.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import fs from "async-file";
+import path from "path";
+import { CommandExists } from "../lib/command-exists";
+import "./editor";
+
+export default class Xcode implements Editor {
+  private commandExists = new CommandExists();
+
+  public get name(): string {
+    return "xed";
```
Comment (alanhamlett) [inline src/editors/xcode.ts]: Typo, should be `Xcode`. Also, remove `displayName`.

File: src/editors/editor.ts (hunk 1 above)
Comment (gandarez) [reply src/editors/editor.ts]: I think name can be accepted as a kind of `key` and `displayName` to friendly show the editr name.

File: src/editors/xcode.ts (hunk 2 above)
Comment (gandarez) [reply src/editors/xcode.ts]: `name` could be used for the `key`

File: src/editors/editor.ts (hunk 1 above)
Comment (alanhamlett) [reply src/editors/editor.ts]: I was thinking `name` should be called `id` or `key` and implemented on `Editor` parent class by replacing whitespace and lowercasing `name`?

Comment (gandarez) [reply src/editors/editor.ts]: If the `key` property gets `displayName` and remove whitespace and lowercasing?

**Pull request merged**